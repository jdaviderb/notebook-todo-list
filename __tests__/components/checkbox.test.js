import React, { Component } from 'react';
import { shallow } from 'enzyme';
import Checkbox from '../../app/components/checkbox';

describe('Component: Checkbox', () => {

  it('renders correctly', () => {
    const wrapper = shallow(<Checkbox />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render correctyle label', () => {
    const wrapper = shallow(<Checkbox />);
    const expectedLabel = 'Venezuela';
    wrapper.setProps({ label: expectedLabel });
    const label = wrapper.find('Text').children().text();
    expect(label).toEqual(expectedLabel);
  });

  it('render a stroke when is active ', () => {
    const wrapper = shallow(<Checkbox />);
    wrapper.setProps({ active: true });
    const hasStroke = wrapper.find('Stroke').exists();
    expect(hasStroke).toBeTruthy();
  });

  it('render a Checkbox without stroke', () => {
    const wrapper = shallow(<Checkbox />);
    wrapper.setProps({ active: false });
    const hasStroke = wrapper.find('Stroke').exists();
    expect(hasStroke).toBeFalsy();
  });

});