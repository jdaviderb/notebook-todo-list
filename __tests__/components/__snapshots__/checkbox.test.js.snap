// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Checkbox renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Checkbox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TouchableOpacity
        activeOpacity={0.2}
        onLongPress={undefined}
        onPress={undefined}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Component
          style={
            Array [
              Object {
                "borderStyle": "dashed",
                "borderWidth": 1.5,
                "height": 40,
                "width": 40,
              },
              Object {
                "borderColor": "black",
              },
            ]
          }
        />
        <Component
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "rgba(0,0,0,.1)",
              "left": 10,
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontFamily": "Cheveuxdange",
                "fontSize": 20,
              }
            }
          />
        </Component>
      </TouchableOpacity>,
      "style": Object {
        "flexDirection": "row",
        "margin": 10,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": Array [
          <Component
            style={
              Array [
                Object {
                  "borderStyle": "dashed",
                  "borderWidth": 1.5,
                  "height": 40,
                  "width": 40,
                },
                Object {
                  "borderColor": "black",
                },
              ]
            }
          />,
          <Component
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "rgba(0,0,0,.1)",
                "left": 10,
                "width": "100%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "black",
                  "fontFamily": "Cheveuxdange",
                  "fontSize": 20,
                }
              }
            />
          </Component>,
        ],
        "onLongPress": undefined,
        "onPress": undefined,
        "style": Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "style": Array [
              Object {
                "borderStyle": "dashed",
                "borderWidth": 1.5,
                "height": 40,
                "width": 40,
              },
              Object {
                "borderColor": "black",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "black",
                  "fontFamily": "Cheveuxdange",
                  "fontSize": 20,
                }
              }
            />,
            "style": Object {
              "borderBottomWidth": 1,
              "borderColor": "rgba(0,0,0,.1)",
              "left": 10,
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": undefined,
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "black",
                "fontFamily": "Cheveuxdange",
                "fontSize": 20,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TouchableOpacity
          activeOpacity={0.2}
          onLongPress={undefined}
          onPress={undefined}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Component
            style={
              Array [
                Object {
                  "borderStyle": "dashed",
                  "borderWidth": 1.5,
                  "height": 40,
                  "width": 40,
                },
                Object {
                  "borderColor": "black",
                },
              ]
            }
          />
          <Component
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "rgba(0,0,0,.1)",
                "left": 10,
                "width": "100%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "black",
                  "fontFamily": "Cheveuxdange",
                  "fontSize": 20,
                }
              }
            />
          </Component>
        </TouchableOpacity>,
        "style": Object {
          "flexDirection": "row",
          "margin": 10,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": Array [
            <Component
              style={
                Array [
                  Object {
                    "borderStyle": "dashed",
                    "borderWidth": 1.5,
                    "height": 40,
                    "width": 40,
                  },
                  Object {
                    "borderColor": "black",
                  },
                ]
              }
            />,
            <Component
              style={
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "rgba(0,0,0,.1)",
                  "left": 10,
                  "width": "100%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "Cheveuxdange",
                    "fontSize": 20,
                  }
                }
              />
            </Component>,
          ],
          "onLongPress": undefined,
          "onPress": undefined,
          "style": Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "style": Array [
                Object {
                  "borderStyle": "dashed",
                  "borderWidth": 1.5,
                  "height": 40,
                  "width": 40,
                },
                Object {
                  "borderColor": "black",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "Cheveuxdange",
                    "fontSize": 20,
                  }
                }
              />,
              "style": Object {
                "borderBottomWidth": 1,
                "borderColor": "rgba(0,0,0,.1)",
                "left": 10,
                "width": "100%",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": undefined,
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "black",
                  "fontFamily": "Cheveuxdange",
                  "fontSize": 20,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
